### PR TITLE
Created Login Component and Training list Component

### PR DESCRIPTION
**Pull request by ShahinaKP**
Hi @nishintn ,

Please review and merge this branch having following implementations:
1. Login Component
2. Login service using users json data
3. Training List component
4. Service for populating training list

### CODE REVIEW

File: app/app.component.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { Component} from '@angular/core';
+import { Component} from "@angular/core";
 @Component({
-  selector: 'my-app',  
-  template: '<h1>Login</h1>'
+  selector: "my-app",
```
Comment (mad-nishin) [inline app/app.component.ts]: Rename the name.

File: app/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ 
+	doLogin(value) {
+		this.loginService.getUsers()
+		.subscribe(
+            data => {
+            	if(this.chkUser(value, data)) {
+            		this.router.navigate(['/trainings']);
+            	}
+            	else {
+            		alert('Enter Valid Credentials');
```
Comment (mad-nishin) [inline app/login/login.component.ts]: No alert / console stms.

File: app/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  styleUrls:  ["app/login/login.component.css"]
+})
+export class LoginComponent {
+	loginForm :FormGroup;
+	user: {};
+
+    constructor(
+    	public fb: FormBuilder,
+    	private router: Router,
+    	private loginService: LoginService
```
Comment (mad-nishin) [inline app/login/login.component.ts]: Why defining service as private?

File: app/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	) {
+    	this.loginForm = this.fb.group({
+		   	username: ["", Validators.required],
+	    	password: ["", Validators.required];
+	    });
+	}
+ 
+	doLogin(value) {
+		this.loginService.getUsers()
+		.subscribe(
```
Comment (mad-nishin) [inline app/login/login.component.ts:29]: What does subscribe do?

File: app/login/login.component.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            data => {
```
Comment (mad-nishin) [inline app/login/login.component.ts:30]: => ??

File: app/login/login.service.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class LoginService {
+  constructor(private http: Http) { }
```
Comment (mad-nishin) [inline app/login/login.service.ts:8]: private??

File: app/login/login.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            	}
+            },
+            error => {
+            	//Handle Error here
+            }
+        );
+	}
+
+	chkUser(userValue, users) {
+		this.user = users.filter(element => {
```
Comment (mad-nishin) [inline app/login/login.component.ts:48]: => ??

File: app/shared/footer/footer.component.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { Component} from "@angular/core";
+
+@Component({
+  selector: "my-footer",
```
Comment (mad-nishin) [inline app/shared/footer/footer.component.ts]: remane my

File: app/shared/header/header.component.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import { Component} from "@angular/core";
+
+@Component({
+  selector: "my-header",
```
Comment (mad-nishin) [inline app/shared/header/header.component.ts]: rename my

File: app/trainings/traininglist.component.html [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<div class="trainings-wrapper">
+	<div class="training-wrapper" *ngFor="let trainingData of trainings">
```
Comment (mad-nishin) [inline app/trainings/traininglist.component.html:2]: let?

File: app/app.module.ts [hunk 11]
```diff
@@ -1,11 +1,27 @@
...
+import { FooterModule} from "./shared/footer/footer.module";
+
+import { LoginModule} from "./login/login.module";
+
+import { TrainingListModule} from "./trainings/traininglist.module";
+
+import { HeaderComponent} from "./shared/header/header.component";
+import { FooterComponent} from "./shared/footer/footer.component";
+
+import { LoginService } from "./login/login.service";
```
Comment (mad-nishin) [inline app/app.module.ts]: Do we need to import service here?

**Comment by mad-nishin** [comment]
@ShahinaKP,
I have added few comments.  Please correct and revert.

File: app/app.component.ts (hunk 1 above)
Comment (ShahinaKP) [reply app/app.component.ts]: Renamed to "trainings-app"

File: app/login/login.component.ts (hunk 2 above)
Comment (ShahinaKP) [reply app/login/login.component.ts]: Removed alert and added error messages

File: app/login/login.component.ts (hunk 4 above)
Comment (ShahinaKP) [reply app/login/login.component.ts:29]: .subscribe is the function that actually executes the observable.Here 'http' service response is handled using .subscribe. It takes three callback parameters as follow:
.subscribe(success, failure, complete);

File: app/login/login.component.ts (hunk 3 above)
Comment (ShahinaKP) [reply app/login/login.component.ts]: private variables are visible within the class only and during the compiling these are hide while using other classes and thus performance improves

File: app/login/login.service.ts (hunk 6 above)
Comment (ShahinaKP) [reply app/login/login.service.ts:8]: private variables are visible within the class only and are hidden while not in use and thus performance improves.

File: app/shared/footer/footer.component.ts (hunk 8 above)
Comment (ShahinaKP) [reply app/shared/footer/footer.component.ts]: Renamed

File: app/shared/header/header.component.ts (hunk 9 above)
Comment (ShahinaKP) [reply app/shared/header/header.component.ts]: Renamed

File: app/trainings/traininglist.component.html (hunk 10 above)
Comment (ShahinaKP) [reply app/trainings/traininglist.component.html:2]: let is a type of variable declaration and these types are hidden while not in use and thus performance improves.

File: app/app.module.ts (hunk 11 above)
Comment (ShahinaKP) [reply app/app.module.ts]: No need. Its moved to LoginModule.

File: app/login/login.component.ts (hunk 7 above)
Comment (ShahinaKP) [reply app/login/login.component.ts:48]: => defines a function type where the arguments are to the left of the => and the return type is on the right

File: app/login/login.component.ts (hunk 5 above)
Comment (ShahinaKP) [reply app/login/login.component.ts:30]: => defines a function type where the arguments are to the left of the => and the return type is on the right

**Comment by ShahinaKP** [comment]
@nishintn 
Corrected the code based on review comments and pushed. Please review and merge the branch.

**Pull request merged**